### ISSUE TITLE
Missing relevant lines in Flow errors in 0.15

### ISSUE THREAD

**Issue by danvk** [facebook/flow#800]
Error messages have made great strides in Flow-land! But there's still room for improvement. I've just run into an example of a tough-to-debug missing line number in an error today that I wanted to share.

Here's my repo in a broken state:

```
$ git clone https://github.com/hammerlab/pileup.js.git
$ git reset --hard 7ae00b9
$ npm install
$ flow status

/Users/danvk/github/pileup.js/src/main/SamRead.js:40:11,21: property `buffer`
Property not found in
/Users/danvk/github/pileup.js/src/main/Alignment.js:23:25,37:1: object type

/Users/danvk/github/pileup.js/src/main/SamRead.js:41:11,23: property `offset`
Property not found in
/Users/danvk/github/pileup.js/src/main/Alignment.js:23:25,37:1: object type

/Users/danvk/github/pileup.js/src/main/SamRead.js:44:10,15: property `refID`
Property not found in
/Users/danvk/github/pileup.js/src/main/Alignment.js:23:25,37:1: object type

/Users/danvk/github/pileup.js/src/main/SamRead.js:46:10,15: property `l_seq`
Property not found in
/Users/danvk/github/pileup.js/src/main/Alignment.js:23:25,37:1: object type

/Users/danvk/github/pileup.js/src/main/SamRead.js:49:11,16: property `_full`
Property not found in
/Users/danvk/github/pileup.js/src/main/Alignment.js:23:25,37:1: object type

/Users/danvk/github/pileup.js/src/main/SamRead.js:50:10,15: property `_seq`
Property not found in
/Users/danvk/github/pileup.js/src/main/Alignment.js:23:25,37:1: object type

/Users/danvk/github/pileup.js/src/main/SamRead.js:51:15,36: property `_interval`
Property not found in
/Users/danvk/github/pileup.js/src/main/Alignment.js:23:25,37:1: object type

/Users/danvk/github/pileup.js/src/main/SamRead.js:78:3,81:3: property `_getJDataView`
Property not found in
/Users/danvk/github/pileup.js/src/main/Alignment.js:23:25,37:1: object type

/Users/danvk/github/pileup.js/src/main/SamRead.js:99:3,101:3: property `getFlag`
Property not found in
/Users/danvk/github/pileup.js/src/main/Alignment.js:23:25,37:1: object type

/Users/danvk/github/pileup.js/src/main/SamRead.js:108:3,114:3: property `getFull`
Property not found in
/Users/danvk/github/pileup.js/src/main/Alignment.js:23:25,37:1: object type

/Users/danvk/github/pileup.js/src/main/SamRead.js:156:3,158:3: property `getCigarString`
Property not found in
/Users/danvk/github/pileup.js/src/main/Alignment.js:23:25,37:1: object type

/Users/danvk/github/pileup.js/src/main/SamRead.js:160:3,162:3: property `getQualPhred`
Property not found in
/Users/danvk/github/pileup.js/src/main/Alignment.js:23:25,37:1: object type

/Users/danvk/github/pileup.js/src/main/SamRead.js:209:3,220:3: property `debugString`
Property not found in
/Users/danvk/github/pileup.js/src/main/Alignment.js:23:25,37:1: object type

Found 13 errors
```

The error messages mention `SamRead.js` and `Alignment.js`. All properties & methods on `SamRead` which aren't declared in the `Alignment` exported type are being flagged as errors.

It's really unclear how Flow is discovering a connection between this class and type, and why it's getting it backwards. (I expect `SamRead` to implement all methods in `Alignment`, plus some of its own.)

After puzzling it over, I found [the fix][1] in another file, `pileuputils.js`:

```
-function getOpInfo(read: SamRead, referenceSource: Object): OpInfo {
+function getOpInfo(read: Alignment, referenceSource: Object): OpInfo {
```

```
$ git reset --hard cc4bf8c1
$ flow status
No errors!
```

I believe the chain is that `pileuputils.getOpInfo` was being called with an `Alignment` in `PileupTrack.js`, but was declared to take a `SamRead`. Hence Flow was explaining why an `Alignment` is not a `SamRead`, but without the context of why it was expecting this to be the case.

A good error message for this issue would have started by mentioning these two relevant lines:
```
PileupTrack.js:265
pileuputils.js:133
```

I'd be happy to distill a more isolated repro if there's interest in improving this error.

[1]: https://github.com/hammerlab/pileup.js/commit/cc4bf8c1458bc919233ed08e56da1b6297740ff8

**Comment by samwgoldman** [comment]
Thanks for the feedback! Examples like this are definitely helpful and we'd like to improve error messages moving forward.

That said, for now you can pass the `--traces` argument to Flow and get some more error details. These details can be hard to parse, but will absolutely point you to more related lines. I know it has helped me find the root cause in some tricky situations before.

**Comment by danvk** [comment]
This got better with Flow 0.17.0. The error now looks like this:

```
src/main/SamRead.js:40
 40:   buffer: ArrayBuffer;
               ^^^^^^^^^^^ property `buffer`. Property not found in
265:         opInfo = getOpInfo(read, referenceSource);
                                ^^^^ object type. See: src/main/PileupTrack.js:265
```

so at least `PileupTrack.js:265` is mentioned. I'll close this out.

**Comment by samwgoldman** [comment]
Nice! You can thank @mroch for this improvement (I think).
